lib/mayaHydra/hydraExtensions/mayaUtils.cpp

```diff
@@ -184,4 +184,15 @@ MObject GetShadingGroupFromShader(const MObject& shader)
     return shadingGroup;
 }
 
+bool IsDagPathAnArnoldSkyDomeLight(const MDagPath& dagPath) 
+{ 
+    static const std::string _aiSkyDomeLight("aiSkyDomeLight");
+
+    if (! dagPath.isValid()) return false;
+
+    const std::string fpName = dagPath.fullPathName().asChar();
+    return (fpName.find(_aiSkyDomeLight) != std::string::npos);
```

Comment:
Shouldn't we be using the type of the node, rather than the node name?  What if the user renames the dome light?

And in any case, to me this immediately brings up the design principle of trying to replace type-based conditionals with virtual functions.  Is there a way we can make this more generic so that we can easily add in the same support for other renderers?